File: src/python/_0001_Two_Sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        hashMap = {}
        for i, val in enumerate(nums):
            diff = target - val
            if diff in hashMap:
                return [hashMap[diff], i]
            hashMap[val] = i

File: src/_0001_Two_Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seenMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (seenMap.containsKey(nums[i]) == true)
                return new int[] { seenMap.get(nums[i]), i };
            seenMap.put(target - nums[i], i);
        }

        return null;
    }
}
